### DISCUSSION TITLE
How to rotate TextBlock Control？

### DISCUSSION THREAD

**Discussion by sparrow2015** [AvaloniaUI/Avalonia#14910] [Q&A]
I want to rotate TextBlock and display text vertical. Like below:
![image](https://github.com/AvaloniaUI/Avalonia/assets/16752460/9c5c7fbe-b7c7-4733-9276-2b616c647eb8)
What should I do to set TextBlock control?

**Comment by sparrow2015** [comment]
I have found resolution, add code below:
`textBlock.RenderTransform = new RotateTransform(angle: 270);`

**Reply by timunie** [reply]
RenderTransform works but may have some drwabacks in your case (layout will not be updated). You can use LayoutTransformControl if you need layout to update as well: https://docs.avaloniaui.net/docs/reference/controls/layouttransformcontrol

**Reply by sparrow2015** [reply]
Thank you very much, i will try LayoutTransformControl.

**Reply by AscendLiu** [reply]
> RenderTransform works but may have some drwabacks in your case (layout will not be updated). You can use LayoutTransformControl if you need layout to update as well: https://docs.avaloniaui.net/docs/reference/controls/layouttransformcontrol

Hi，How can I directly draw a rotated text using DrawingContext

**Reply by maxkatz6** [reply]
You can use matrix transformations, and set them via DrawingContext.PushTransform

**Reply by AscendLiu** [reply]
``` C#
    private void DrawYAxis(DrawingContext context, Rect rect)
    {
        if (string.IsNullOrWhiteSpace(YAxisLabel)) return;
        FormattedText text = new(YAxisLabel,
            CultureInfo.CurrentCulture,
            FlowDirection.LeftToRight, 
            Typeface.Default,
            12,
            Brushes.Black);
        Point valueTextPosition = new(rect.X + 8, rect.Y + (rect.Height - text.Width) / 2 + text.Width);
        var translateModifier = context.PushTransform(Avalonia.Matrix.CreateTranslation(new Avalonia.Vector(valueTextPosition.X, valueTextPosition.Y)));
        var rotationMatrix = Avalonia.Matrix.CreateRotation(-1.570796);
        var rotationModifier = context.PushTransform(rotationMatrix);
        context.DrawText(text, new Point(0,0));

        rotationModifier.Dispose();
        translateModifier.Dispose();
    }
```

![图片](https://github.com/user-attachments/assets/a746b84f-dd25-4384-8d77-b7dd30c277f1)

I succeeded
